### DISCUSSION TITLE
Make `PathParams` type `Record<string, string | ReadonlyArray<string>>`?

### DISCUSSION THREAD

**Discussion by stefanprobst** [mswjs/msw#1017] [Ideas]
hi, would you consider changing the `PathParams` type from [`Record<string, string | string[]>`](https://github.com/mswjs/msw/blob/93d2226d2ec5d5e5d9dfb6036e4a36e0633cbab6/src/utils/matching/matchRequestUrl.ts#L6) to `Record<string, string | ReadonlyArray<string>>`?

**Comment by kettanaito** [comment]
Hey, @stefanprobst. Thank you for the suggestion.

What benefits do you think it'd provide if we switched to `ReadonlyArray`?

**Reply by stefanprobst** [reply]
afaiu it is ok to pass `Array` where `ReadonlyArray` is expected, but not the other way around. so i have something similar to the following, which does not play nice with the current `PathParams` from `msw`:

https://www.typescriptlang.org/play?#code/C4TwDgpgBACghsAFvATnAtgZygXigJQgGMB7FAEwB5NgUBLAOwHMAaKG+5qAH3dsaYBtALoA+AFChIsBMjhos+AIK4CxMlQ4C2Wrr0JxyJBgBsQSlGhDV+zURPGNgEFADM4RaAFkQ8JKgxsCAAPZwZybEJSChtOVj44njVDYzMLK1iBeygAb3EoAqg6cgAuBIFxAF9xcVIGGigweUCynz85BWw8HKLSqAByAEZ+qGrxcmITeWhXAFcGImA6YygiRGIAawAKJs6y9oCsAEoygDcSYtr1om3dwKOaiaIplBn5xeWGVeuN5R3mrD7WSHTDKE5Qc6XNabP53Y7iIA

**Reply by kettanaito** [reply]
Got it. Thank you for sharing your use case.
I see no harm in annotating path parameters as `ReadonlyArray`, given that you cannot, in fact, change that array. Would you be interested in opening a pull request with this change?

**Reply by stefanprobst** [reply]
> Would you be interested in opening a pull request with this change?

sure! thanks!
